# Bellabeat Data Analysis Project

## Background

Bellabeat is a high-tech manufactuer focusng of women health since 2013. Their well-designed products includes the Bellabeat app, wellness trackers Leaf, Time and Spring, as well as the Bellabeat membership, which forms a strong ecosystem to empower women with the knowledge abour their own health and habits. Now, Bellabeat is seeking more opportunities to grow by learning the current smart device market. 

### Business Objectives
The key task of this report is to analyze FitBit Fitness Tracker Data to gain insights into how consumers are using the their smart devices and give high-level recommendation for Bellabeat's marketing stretegy. 

### Key Stakeholders
* Urška Sršen: Bellabeat’s cofounder and Chief Creative Officer
* Sando Mur: Mathematician, Bellabeat’s cofounder and key member of the Bellabeat executive team
* Bellabeat marketing analytics team: A team of data analysts guiding Bellabeat's marketing strategy.


## Data Preparation

### Import packages and dataset

In [144]:
# import packages 
import numpy as np 
import pandas as pd 
import matplotlib as plt 
import datetime as dt 

In [145]:
# import data 
activity = pd.read_csv("/Users/yuefei/Documents/Code/Bellabeat-data-analysis-project/Data/Fitabase Data 4.12.16-5.12.16/dailyActivity_merged.csv")
calories = pd.read_csv("/Users/yuefei/Documents/Code/Bellabeat-data-analysis-project/Data/Fitabase Data 4.12.16-5.12.16/dailyCalories_merged.csv")
intensities = pd.read_csv("/Users/yuefei/Documents/Code/Bellabeat-data-analysis-project/Data/Fitabase Data 4.12.16-5.12.16/dailyIntensities_merged.csv")
steps = pd.read_csv("/Users/yuefei/Documents/Code/Bellabeat-data-analysis-project/Data/Fitabase Data 4.12.16-5.12.16/dailySteps_merged.csv")
sleep = pd.read_csv("/Users/yuefei/Documents/Code/Bellabeat-data-analysis-project/Data/Fitabase Data 4.12.16-5.12.16/sleepDay_merged.csv")
weight = pd.read_csv("/Users/yuefei/Documents/Code/Bellabeat-data-analysis-project/Data/Fitabase Data 4.12.16-5.12.16/weightLogInfo_merged.csv")

#### Daily Activity

In [146]:
# Have a look at the first 5 rows and the basic information
activity.head()

Id ActivityDate  TotalSteps  TotalDistance  TrackerDistance  \
0  1503960366    4/12/2016       13162           8.50             8.50   
1  1503960366    4/13/2016       10735           6.97             6.97   
2  1503960366    4/14/2016       10460           6.74             6.74   
3  1503960366    4/15/2016        9762           6.28             6.28   
4  1503960366    4/16/2016       12669           8.16             8.16   

   LoggedActivitiesDistance  VeryActiveDistance  ModeratelyActiveDistance  \
0                       0.0                1.88                      0.55   
1                       0.0                1.57                      0.69   
2                       0.0                2.44                      0.40   
3                       0.0                2.14                      1.26   
4                       0.0                2.71                      0.41   

   LightActiveDistance  SedentaryActiveDistance  VeryActiveMinutes  \
0                 6.06                      0.0                 25   
1                 4.71                      0.0                 21   
2                 3.91                      0.0                 30   
3                 2.83                      0.0                 29   
4                 5.04                      0.0                 36   

   FairlyActiveMinutes  LightlyActiveMinutes  SedentaryMinutes  Calories  
0                   13                   328               728      1985  
1                   19                   217               776      1797  
2                   11                   181              1218      1776  
3                   34                   209               726      1745  
4                   10                   221               773      1863

In [147]:
activity.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 940 entries, 0 to 939
Data columns (total 15 columns):
 #   Column                    Non-Null Count  Dtype  
---  ------                    --------------  -----  
 0   Id                        940 non-null    int64  
 1   ActivityDate              940 non-null    object 
 2   TotalSteps                940 non-null    int64  
 3   TotalDistance             940 non-null    float64
 4   TrackerDistance           940 non-null    float64
 5   LoggedActivitiesDistance  940 non-null    float64
 6   VeryActiveDistance        940 non-null    float64
 7   ModeratelyActiveDistance  940 non-null    float64
 8   LightActiveDistance       940 non-null    float64
 9   SedentaryActiveDistance   940 non-null    float64
 10  VeryActiveMinutes         940 non-null    int64  
 11  FairlyActiveMinutes       940 non-null    int64  
 12  LightlyActiveMinutes      940 non-null    int64  
 13  SedentaryMinutes          940 non-null    int64  
 14  Calories  

Check if there is any missing data in daily activity. Turns out no missing data in the activity table. 

In [148]:
# checking null values
missing_values = activity.isnull().sum()
missing_values

Id                          0
ActivityDate                0
TotalSteps                  0
TotalDistance               0
TrackerDistance             0
LoggedActivitiesDistance    0
VeryActiveDistance          0
ModeratelyActiveDistance    0
LightActiveDistance         0
SedentaryActiveDistance     0
VeryActiveMinutes           0
FairlyActiveMinutes         0
LightlyActiveMinutes        0
SedentaryMinutes            0
Calories                    0
dtype: int64

Check the user number in this dataset and their id

In [149]:
# checking the sample size and their unique user number
unique_id = np.unique(activity.Id)
unique_user_number = np.count_nonzero(unique_id)
unique_id, unique_user_number

(array([1503960366, 1624580081, 1644430081, 1844505072, 1927972279,
        2022484408, 2026352035, 2320127002, 2347167796, 2873212765,
        3372868164, 3977333714, 4020332650, 4057192912, 4319703577,
        4388161847, 4445114986, 4558609924, 4702921684, 5553957443,
        5577150313, 6117666160, 6290855005, 6775888955, 6962181067,
        7007744171, 7086361926, 8053475328, 8253242879, 8378563200,
        8583815059, 8792009665, 8877689391]),
 33)

The activity data can be observed: 
* The activity table has 15 columns and 940 rows
* No missing data
* The sample size is 33, which is quite small. 
* The issue here is that date is marked as `object`. 

Meanwhile, I would like to check the other tables to see whether the data type issue is common. 

#### Calories

In [150]:
# checking calories table
calories.head()

Id ActivityDay  Calories
0  1503960366   4/12/2016      1985
1  1503960366   4/13/2016      1797
2  1503960366   4/14/2016      1776
3  1503960366   4/15/2016      1745
4  1503960366   4/16/2016      1863

In [151]:
calories.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 940 entries, 0 to 939
Data columns (total 3 columns):
 #   Column       Non-Null Count  Dtype 
---  ------       --------------  ----- 
 0   Id           940 non-null    int64 
 1   ActivityDay  940 non-null    object
 2   Calories     940 non-null    int64 
dtypes: int64(2), object(1)
memory usage: 22.2+ KB


In [152]:
calories_missing_values = calories.isnull().sum()
calories_missing_values

Id             0
ActivityDay    0
Calories       0
dtype: int64

#### Intensities

In [153]:
# Intensities table
intensities.head()

Id ActivityDay  SedentaryMinutes  LightlyActiveMinutes  \
0  1503960366   4/12/2016               728                   328   
1  1503960366   4/13/2016               776                   217   
2  1503960366   4/14/2016              1218                   181   
3  1503960366   4/15/2016               726                   209   
4  1503960366   4/16/2016               773                   221   

   FairlyActiveMinutes  VeryActiveMinutes  SedentaryActiveDistance  \
0                   13                 25                      0.0   
1                   19                 21                      0.0   
2                   11                 30                      0.0   
3                   34                 29                      0.0   
4                   10                 36                      0.0   

   LightActiveDistance  ModeratelyActiveDistance  VeryActiveDistance  
0                 6.06                      0.55                1.88  
1                 4.71                      0.69                1.57  
2                 3.91                      0.40                2.44  
3                 2.83                      1.26                2.14  
4                 5.04                      0.41                2.71

In [154]:
intensities.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 940 entries, 0 to 939
Data columns (total 10 columns):
 #   Column                    Non-Null Count  Dtype  
---  ------                    --------------  -----  
 0   Id                        940 non-null    int64  
 1   ActivityDay               940 non-null    object 
 2   SedentaryMinutes          940 non-null    int64  
 3   LightlyActiveMinutes      940 non-null    int64  
 4   FairlyActiveMinutes       940 non-null    int64  
 5   VeryActiveMinutes         940 non-null    int64  
 6   SedentaryActiveDistance   940 non-null    float64
 7   LightActiveDistance       940 non-null    float64
 8   ModeratelyActiveDistance  940 non-null    float64
 9   VeryActiveDistance        940 non-null    float64
dtypes: float64(4), int64(5), object(1)
memory usage: 73.6+ KB


In [155]:
intensities_missing_values = intensities.isnull().sum()
intensities_missing_values

Id                          0
ActivityDay                 0
SedentaryMinutes            0
LightlyActiveMinutes        0
FairlyActiveMinutes         0
VeryActiveMinutes           0
SedentaryActiveDistance     0
LightActiveDistance         0
ModeratelyActiveDistance    0
VeryActiveDistance          0
dtype: int64

#### Steps

In [156]:
# steps table
steps.head()

Id ActivityDay  StepTotal
0  1503960366   4/12/2016      13162
1  1503960366   4/13/2016      10735
2  1503960366   4/14/2016      10460
3  1503960366   4/15/2016       9762
4  1503960366   4/16/2016      12669

In [157]:
steps.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 940 entries, 0 to 939
Data columns (total 3 columns):
 #   Column       Non-Null Count  Dtype 
---  ------       --------------  ----- 
 0   Id           940 non-null    int64 
 1   ActivityDay  940 non-null    object
 2   StepTotal    940 non-null    int64 
dtypes: int64(2), object(1)
memory usage: 22.2+ KB


In [158]:
steps_missing_values = steps.isnull().sum()
steps_missing_values

Id             0
ActivityDay    0
StepTotal      0
dtype: int64

#### Sleep

In [159]:
# sleep
sleep.head()

Id               SleepDay  TotalSleepRecords  TotalMinutesAsleep  \
0  1503960366  4/12/2016 12:00:00 AM                  1                 327   
1  1503960366  4/13/2016 12:00:00 AM                  2                 384   
2  1503960366  4/15/2016 12:00:00 AM                  1                 412   
3  1503960366  4/16/2016 12:00:00 AM                  2                 340   
4  1503960366  4/17/2016 12:00:00 AM                  1                 700   

   TotalTimeInBed  
0             346  
1             407  
2             442  
3             367  
4             712

In [160]:
sleep.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 413 entries, 0 to 412
Data columns (total 5 columns):
 #   Column              Non-Null Count  Dtype 
---  ------              --------------  ----- 
 0   Id                  413 non-null    int64 
 1   SleepDay            413 non-null    object
 2   TotalSleepRecords   413 non-null    int64 
 3   TotalMinutesAsleep  413 non-null    int64 
 4   TotalTimeInBed      413 non-null    int64 
dtypes: int64(4), object(1)
memory usage: 16.3+ KB


In [161]:
sleep_missing_values = sleep.isnull().sum()
sleep_missing_values

Id                    0
SleepDay              0
TotalSleepRecords     0
TotalMinutesAsleep    0
TotalTimeInBed        0
dtype: int64

#### Weight

In [162]:
# weight
weight.head()

Id                   Date    WeightKg  WeightPounds   Fat  \
0  1503960366   5/2/2016 11:59:59 PM   52.599998    115.963147  22.0   
1  1503960366   5/3/2016 11:59:59 PM   52.599998    115.963147   NaN   
2  1927972279   4/13/2016 1:08:52 AM  133.500000    294.317120   NaN   
3  2873212765  4/21/2016 11:59:59 PM   56.700001    125.002104   NaN   
4  2873212765  5/12/2016 11:59:59 PM   57.299999    126.324875   NaN   

         BMI  IsManualReport          LogId  
0  22.650000            True  1462233599000  
1  22.650000            True  1462319999000  
2  47.540001           False  1460509732000  
3  21.450001            True  1461283199000  
4  21.690001            True  1463097599000

In [163]:
weight.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 67 entries, 0 to 66
Data columns (total 8 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   Id              67 non-null     int64  
 1   Date            67 non-null     object 
 2   WeightKg        67 non-null     float64
 3   WeightPounds    67 non-null     float64
 4   Fat             2 non-null      float64
 5   BMI             67 non-null     float64
 6   IsManualReport  67 non-null     bool   
 7   LogId           67 non-null     int64  
dtypes: bool(1), float64(4), int64(2), object(1)
memory usage: 3.9+ KB


In [164]:
weight_missing_values = weight.isnull().sum()
weight_missing_values

Id                 0
Date               0
WeightKg           0
WeightPounds       0
Fat               65
BMI                0
IsManualReport     0
LogId              0
dtype: int64

After checking all the tables, it is confirmed that all the date are stored as object. 

The following date transformation will be performed: 
1. All the date is converted from `object` to `datetime`
2. The date is formatted to ``dd-mm-yyyy`` as commonly used in Australia
3. Date with daytime like `5/2/2016 11:59:59 PM` will be split into two columns

As the Calories, Intensities, Steps data are included in the daily activity table,  these three table will not be used in the following steps. We will mainly look at the daily activity table.

### Data Transformation

In [165]:
# convert the activity data time
activity["ActivityDate"] = pd.to_datetime(activity["ActivityDate"],format="%m/%d/%Y")
activity["DayofWeek"] = activity["ActivityDate"].dt.day_name()

# add up total activity time
activity["TotalActiveMinutes"] = activity["FairlyActiveMinutes"] + activity["LightActiveDistance"] + activity["VeryActiveMinutes"] + activity["SedentaryMinutes"]

# active time in hour
activity["TotalActiveHours"] = round(activity["TotalActiveMinutes"]/60, 2)
activity.head()

Id ActivityDate  TotalSteps  TotalDistance  TrackerDistance  \
0  1503960366   2016-04-12       13162           8.50             8.50   
1  1503960366   2016-04-13       10735           6.97             6.97   
2  1503960366   2016-04-14       10460           6.74             6.74   
3  1503960366   2016-04-15        9762           6.28             6.28   
4  1503960366   2016-04-16       12669           8.16             8.16   

   LoggedActivitiesDistance  VeryActiveDistance  ModeratelyActiveDistance  \
0                       0.0                1.88                      0.55   
1                       0.0                1.57                      0.69   
2                       0.0                2.44                      0.40   
3                       0.0                2.14                      1.26   
4                       0.0                2.71                      0.41   

   LightActiveDistance  SedentaryActiveDistance  VeryActiveMinutes  \
0                 6.06                      0.0                 25   
1                 4.71                      0.0                 21   
2                 3.91                      0.0                 30   
3                 2.83                      0.0                 29   
4                 5.04                      0.0                 36   

   FairlyActiveMinutes  LightlyActiveMinutes  SedentaryMinutes  Calories  \
0                   13                   328               728      1985   
1                   19                   217               776      1797   
2                   11                   181              1218      1776   
3                   34                   209               726      1745   
4                   10                   221               773      1863   

   DayofWeek  TotalActiveMinutes  TotalActiveHours  
0    Tuesday              772.06             12.87  
1  Wednesday              820.71             13.68  
2   Thursday             1262.91             21.05  
3     Friday              791.83             13.20  
4   Saturday              824.04             13.73

In [166]:
# rename column name
act_new_col_name = {
        "Id":"id", 
        "ActivityDate":"date",
        "TotalSteps":"total_steps", 
        "TotalDistance":"total_dist", 
        "TrackerDistance":"track_dist",
        "LoggedActivitiesDistance":"logged_dist", 
        "VeryActiveDistance":"very_active_dist", 
        "ModeratelyActiveDistance":"moderate_active_dist", 
        "LightActiveDistance":"light_active_dist", 
        "SedentaryActiveDistance":"sedentary_active_dist", 
        "VeryActiveMinutes":"very_active_mins", 
        "FairlyActiveMinutes":"fairly_active_mins", 
        "LightlyActiveMinutes":"lightly_active_mins", 
        "SedentaryMinutes":"sedentary_mins", 
        "Calories":"calories",
        "DayofWeek":"day_of_week", 
        "TotalActiveMinutes":"total_active_mins",
        "TotalActiveHours":"total_active_hours"
        }
   
# reorder the columns
act_new_col_order = [
        "id", "date","day_of_week", "total_steps", "total_dist", 
        "track_dist", "logged_dist", "very_active_dist", 
        "moderate_active_dist", "light_active_dist", "sedentary_active_dist", 
        "very_active_mins", "fairly_active_mins", "lightly_active_mins", 
        "sedentary_mins", "total_active_mins", "total_active_hours", "calories"
        ]

# update the table
activity = activity.rename(columns=act_new_col_name)[act_new_col_order]

#check the updated activity data
activity.head()

id       date day_of_week  total_steps  total_dist  track_dist  \
0  1503960366 2016-04-12     Tuesday        13162        8.50        8.50   
1  1503960366 2016-04-13   Wednesday        10735        6.97        6.97   
2  1503960366 2016-04-14    Thursday        10460        6.74        6.74   
3  1503960366 2016-04-15      Friday         9762        6.28        6.28   
4  1503960366 2016-04-16    Saturday        12669        8.16        8.16   

   logged_dist  very_active_dist  moderate_active_dist  light_active_dist  \
0          0.0              1.88                  0.55               6.06   
1          0.0              1.57                  0.69               4.71   
2          0.0              2.44                  0.40               3.91   
3          0.0              2.14                  1.26               2.83   
4          0.0              2.71                  0.41               5.04   

   sedentary_active_dist  very_active_mins  fairly_active_mins  \
0                    0.0                25                  13   
1                    0.0                21                  19   
2                    0.0                30                  11   
3                    0.0                29                  34   
4                    0.0                36                  10   

   lightly_active_mins  sedentary_mins  total_active_mins  total_active_hours  \
0                  328             728             772.06               12.87   
1                  217             776             820.71               13.68   
2                  181            1218            1262.91               21.05   
3                  209             726             791.83               13.20   
4                  221             773             824.04               13.73   

   calories  
0      1985  
1      1797  
2      1776  
3      1745  
4      1863

In [167]:
# format and split the column into date, time and day of week
sleep["SleepDay"] = pd.to_datetime(sleep["SleepDay"],format="%m/%d/%Y %H:%M:%S %p") # format
sleep["DayofWeek"] = sleep["SleepDay"].dt.day_name() # add new column, assign day of week
sleep["SleepTime"] = sleep["SleepDay"].dt.time # add the time column, slice the time
sleep["SleepDay"] = sleep["SleepDay"].dt.date # slice date

# reorganize the column order
sleep_new_column = ["Id", "SleepDay", "SleepTime", "DayofWeek","TotalSleepRecords", "TotalMinutesAsleep", "TotalTimeInBed"]
sleep = sleep[sleep_new_column] 

In [168]:
sleep.head()

Id    SleepDay SleepTime  DayofWeek  TotalSleepRecords  \
0  1503960366  2016-04-12  12:00:00    Tuesday                  1   
1  1503960366  2016-04-13  12:00:00  Wednesday                  2   
2  1503960366  2016-04-15  12:00:00     Friday                  1   
3  1503960366  2016-04-16  12:00:00   Saturday                  2   
4  1503960366  2016-04-17  12:00:00     Sunday                  1   

   TotalMinutesAsleep  TotalTimeInBed  
0                 327             346  
1                 384             407  
2                 412             442  
3                 340             367  
4                 700             712

In [169]:
# format and split the column into date, time and day of week
weight["Date"] = pd.to_datetime(weight["Date"],format="%m/%d/%Y %I:%M:%S %p")
weight["DayofWeek"] = weight["Date"].dt.day_name() # add new column, assign day of week
weight["Time"] = weight["Date"].dt.time # add the time column, slice the time
weight["Date"] = weight["Date"].dt.date # slice date

# reorganize the column order
weight_new_column = ["Id", "Date", "Time", "DayofWeek", "Fat", "BMI", "IsManualReport", "LogId"]
weight = weight[weight_new_column]
weight.head()

Id        Date      Time  DayofWeek   Fat        BMI  \
0  1503960366  2016-05-02  23:59:59     Monday  22.0  22.650000   
1  1503960366  2016-05-03  23:59:59    Tuesday   NaN  22.650000   
2  1927972279  2016-04-13  01:08:52  Wednesday   NaN  47.540001   
3  2873212765  2016-04-21  23:59:59   Thursday   NaN  21.450001   
4  2873212765  2016-05-12  23:59:59   Thursday   NaN  21.690001   

   IsManualReport          LogId  
0            True  1462233599000  
1            True  1462319999000  
2           False  1460509732000  
3            True  1461283199000  
4            True  1463097599000

In [170]:
# thoughts of the following steps: 
# join tables? 
# frequency on activities or weights
# average sleep time
# 
# recommendations based on the result

# Analysis

In [171]:
# overview on the activity table
activity.describe()

id                           date   total_steps  total_dist  \
count  9.400000e+02                            940    940.000000  940.000000   
mean   4.855407e+09  2016-04-26 06:53:37.021276672   7637.910638    5.489702   
min    1.503960e+09            2016-04-12 00:00:00      0.000000    0.000000   
25%    2.320127e+09            2016-04-19 00:00:00   3789.750000    2.620000   
50%    4.445115e+09            2016-04-26 00:00:00   7405.500000    5.245000   
75%    6.962181e+09            2016-05-04 00:00:00  10727.000000    7.712500   
max    8.877689e+09            2016-05-12 00:00:00  36019.000000   28.030001   
std    2.424805e+09                            NaN   5087.150742    3.924606   

       track_dist  logged_dist  very_active_dist  moderate_active_dist  \
count  940.000000   940.000000        940.000000            940.000000   
mean     5.475351     0.108171          1.502681              0.567543   
min      0.000000     0.000000          0.000000              0.000000   
25%      2.620000     0.000000          0.000000              0.000000   
50%      5.245000     0.000000          0.210000              0.240000   
75%      7.710000     0.000000          2.052500              0.800000   
max     28.030001     4.942142         21.920000              6.480000   
std      3.907276     0.619897          2.658941              0.883580   

       light_active_dist  sedentary_active_dist  very_active_mins  \
count         940.000000             940.000000        940.000000   
mean            3.340819               0.001606         21.164894   
min             0.000000               0.000000          0.000000   
25%             1.945000               0.000000          0.000000   
50%             3.365000               0.000000          4.000000   
75%             4.782500               0.000000         32.000000   
max            10.710000               0.110000        210.000000   
std             2.040655               0.007346         32.844803   

       fairly_active_mins  lightly_active_mins  sedentary_mins  \
count          940.000000           940.000000      940.000000   
mean            13.564894           192.812766      991.210638   
min              0.000000             0.000000        0.000000   
25%              0.000000           127.000000      729.750000   
50%              6.000000           199.000000     1057.500000   
75%             19.000000           264.000000     1229.500000   
max            143.000000           518.000000     1440.000000   
std             19.987404           109.174700      301.267437   

       total_active_mins  total_active_hours     calories  
count         940.000000          940.000000   940.000000  
mean         1029.281245           17.154681  2303.609574  
min             0.010000            0.000000     0.000000  
25%           782.520000           13.040000  1828.500000  
50%          1105.975000           18.430000  2134.000000  
75%          1268.055000           21.130000  2793.250000  
max          1440.000000           24.000000  4900.000000  
std           293.370914            4.889476   718.166862

In [172]:
sleep.describe()

Id  TotalSleepRecords  TotalMinutesAsleep  TotalTimeInBed
count  4.130000e+02         413.000000          413.000000      413.000000
mean   5.000979e+09           1.118644          419.467312      458.639225
std    2.060360e+09           0.345521          118.344679      127.101607
min    1.503960e+09           1.000000           58.000000       61.000000
25%    3.977334e+09           1.000000          361.000000      403.000000
50%    4.702922e+09           1.000000          433.000000      463.000000
75%    6.962181e+09           1.000000          490.000000      526.000000
max    8.792010e+09           3.000000          796.000000      961.000000

In [173]:
weight.describe()

Id       Fat        BMI         LogId
count  6.700000e+01   2.00000  67.000000  6.700000e+01
mean   7.009282e+09  23.50000  25.185224  1.461772e+12
std    1.950322e+09   2.12132   3.066963  7.829948e+08
min    1.503960e+09  22.00000  21.450001  1.460444e+12
25%    6.962181e+09  22.75000  23.959999  1.461079e+12
50%    6.962181e+09  23.50000  24.389999  1.461802e+12
75%    8.877689e+09  24.25000  25.559999  1.462375e+12
max    8.877689e+09  25.00000  47.540001  1.463098e+12

In [ ]:
# activity level on which day of week 


# Recommendation